templates/hiera.yaml.epp

```diff
@@ -33,6 +33,12 @@ hierarchy:
       - "<%= $path %>"
       <%- } -%>
 <% } -%>
+<% if $hr['mapped_paths'] { -%>
+    mapped_paths:
+      - "<%= $hr['mapped_paths'][0] %>"
+      - "<%= $hr['mapped_paths'][1] %>"
+      - "<%= $hr['mapped_paths'][2] %>"
```

Comment:
Why are there three of these instead of iterating over however many might be in the array?

Reply:
Ah.. because that's how it works :)

https://puppet.com/docs/puppet/latest/hiera_config_yaml_5.html#specifying_file_paths

"The mapped_paths key must contain three string elements, in the following order:

    A scope variable that points to a collection of strings.
    The variable name that is mapped to each element of the collection.
    A template where that variable can be used in interpolation expressions."

Reply:
Garrett, just wondering when do you believe it can be pushed up.